source/initial_temperature/anelasticity_temperature.cc

```diff
@@ -0,0 +1,421 @@
+  <http://www.gnu.org/licenses/>.
+*/
+
+#include <aspect/global.h>
+#include <aspect/initial_temperature/anelasticity_temperature.h>
+#include <aspect/material_model/interface.h>
+#include <aspect/initial_composition/interface.h>
+#include <aspect/adiabatic_conditions/interface.h>
+#include <aspect/material_model/anelasticity_temperature.h>
+#include <aspect/simulator_access.h>
+
+#include <boost/lexical_cast.hpp>
+#include <boost/math/tools/minima.hpp>
+
+namespace aspect
+{
+  namespace InitialTemperature
+  {
+    template <int dim>
+    AnelasticVs2T<dim>::AnelasticVs2T ()
+    {}
+
+    template <int dim>
+    void
+    AnelasticVs2T<dim>::initialize ()
+    {
+      Utilities::AsciiDataInitial<dim>::initialize(1);
+    }
+
+     // set up Vs function that Brent minimization operates on
+    template <int dim>
+    double
+    AnelasticVs2T<dim>::
+    fVs (double x,
+         const double      depth,
+         const double      absolute_Vs,
+         const double      mu0,
+         const double      dmudT,
+         const double      dmudP,
+         const double      viscosity_prefactor,
+         const double      activation_energy,
+         const double      activation_volume,
+         const double      solidus_gradient,
+         const bool      use_original_model) const
+    {
+      return std::abs(yamauchi_takei_Vs(x,depth,mu0,dmudT,dmudP,viscosity_prefactor,activation_energy,activation_volume
+                                        ,solidus_gradient,use_original_model)-absolute_Vs);
+    }
+
+    // set up volume change function that Brent minimization operates on
+    template <int dim>
+    double
+    AnelasticVs2T<dim>::
+    fdV (double x, const double bulk_modulus, const double bulk_modulus_pressure_derivative, const double pressure ) const
+    {
+      return std::abs((bulk_modulus*(3./2.)*(std::pow(x,7./3.)-std::pow(x,5./3.))*(1+(((3./4.)*
+                                                                                      (bulk_modulus_pressure_derivative-4))*(std::pow(x,2./3.)-1))))-pressure);
+    }
+
+    // set up initial temperature
+    template <int dim>
+    double
+    AnelasticVs2T<dim>::
+    initial_temperature (const Point<dim> &position) const
+    {
+      // determine depth
+      double depth;
+      depth = this->get_geometry_model().depth(position);
+
+      // declare temperature
+      double temperature;
+
+      // read absolute Vs in from ascii file
+      const double absolute_Vs = Utilities::AsciiDataInitial<dim>::get_data_component(position,0);
+
+      if (depth >= no_perturbation_depth)
+        {
+          // convert absolute Vs into temperature
+          // check if using Yamauchi & Takei 2016 parameterization
+          if (use_yamauchi_takei == true)
+            {
+              // specify Brent algorithm parameters
+              const double a=273;
+              const double b=3273;
+              typedef std::pair<double, double> Result;
```

Comment:
I do not think you need this typedef. See my comment below.

Reply:
You're right, have changed accordingly.